cocos/spine/skeleton-cache.ts

```diff
@@ -347,17 +372,29 @@ class SkeletonCache {
             // No need to create TypedArray next time.
             const animationCache = animationsCache[aniKey];
             if (!animationCache) continue;
-            this._animationPool[`${uuid}#${aniKey}`] = animationCache;
-            animationCache.clear();
+            operate(aniKey, animationCache);
         }
 
+        if (skeletonInfo.skeleton) {
+            spine.wasmUtil.destroySpineSkeleton(skeletonInfo.skeleton);
+        }
         delete this._skeletonCache[uuid];
     }
 
-    public getSkeletonCache (uuid: string, skeletonData: spine.SkeletonData): SkeletonCacheItemInfo {
+    public getSkeletonCache (uuid: string, skeletonData: spine.SkeletonData, skeletonComponent: Skeleton): SkeletonCacheItemInfo {
+        if (SkeletonCache.sharedCache === this) {
+            let sharedInstances = this._sharedCacheMap.get(uuid);
+            if (!sharedInstances) {
+                sharedInstances = new Array<Skeleton>();
+                this._sharedCacheMap.set(uuid, sharedInstances);
+            }
+            if (sharedInstances.indexOf(skeletonComponent) === -1) {
+                sharedInstances.push(skeletonComponent);
+            }
+        }
         let skeletonInfo = this._skeletonCache[uuid];
         if (!skeletonInfo) {
-            const skeleton = null;
+            const skeleton = new spine.Skeleton(skeletonData);
```

Comment:
skeleton is forever `null` in the old code?

Reply:
skeleton will constructor by skeleton component's _instance

Reply:
So we move the `spine.Skeleton` construction code to here?

Reply:
Yes, previous logic multi skeleton component can not share the same object of spine.skeleton